src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -321,6 +351,7 @@ InvokeHostFunctionOpFrame::doApply(AbstractLedgerTxn& ltx, Config const& cfg,
 
         keys.emplace(std::move(lk));
     }
+    metadataSizeBytes += metrics.mLedgerWriteByte;
```

Comment:
It looks like you only validate `metadataSizeBytes` again if the call emitted events. If not, then `mLedgerWriteByte` could cause you to exceed the limit.

Reply:
Nice catch, fixed.